cmd/util/flags/cliflags/config.go

```diff
@@ -27,30 +27,37 @@ func ConfigAutoComplete(cmd *cobra.Command, args []string, toComplete string) ([
 	return completions, cobra.ShellCompDirectiveNoSpace
 }
 
-func NewConfigFlag() *ConfigFlag {
-	viper.Set(RootCommandConfigValues, new(map[string]any))
-	viper.Set(RootCommandConfigFiles, make([]string, 0))
-	return &ConfigFlag{}
-}
-
 const RootCommandConfigFiles = "Root.Command.Config.Files"
 const RootCommandConfigValues = "Root.Command.Config.Values"
 const RootCommandConfigFlags = "Root.Command.Config.Flags"
 
-type Something struct {
-	ConfigFiles  []string
-	ConfigParams map[string]any
+type ConfigFlag struct {
+	Value       string
+	isWriteMode bool
+}
+
+func NewConfigFlag() *ConfigFlag {
+	viper.Set(RootCommandConfigValues, new(map[string]any))
+	viper.Set(RootCommandConfigFiles, make([]string, 0))
+	return &ConfigFlag{isWriteMode: false}
 }
 
-type ConfigFlag struct {
-	Value string
+func NewWriteConfigFlag() *ConfigFlag {
+	viper.Set(RootCommandConfigFiles, "")
+	return &ConfigFlag{isWriteMode: true}
 }
 
 func (cf *ConfigFlag) String() string {
 	return cf.Value
 }
 
 func (cf *ConfigFlag) Set(value string) error {
+	if cf.isWriteMode {
+		// Check if a config file is already set in Viper
+		if viper.GetString(RootCommandConfigFiles) != "" {
+			return fmt.Errorf("single config file can be set")
+		}
+	}
```

Comment:
Is this how we are enforcing that the flag may only be used once with the `config set` command?

Reply:
Correct